plugins/CoreConsole/Commands/GenerateTest.php

```diff
@@ -193,4 +194,86 @@ protected function getTestFilesWhitelist($testType)
             '/tests/Unit/SimpleTest.php'
         );
     }
-}
+
+    private function addTravisJobsIfRequired(OutputInterface $output, $testType, $pluginName)
+    {
+        $travisYmlFile = PIWIK_INCLUDE_PATH . '/plugins/' . $pluginName . '/.travis.yml';
+        if (!file_exists($travisYmlFile)) {
+            $output->writeln("Plugin '$pluginName' has no .travis.yml file, skipping travis job addition.");
+            return;
+        }
+
+        $travisYmlFileContents = file_get_contents($travisYmlFile);
+
+        $testVarEnvVar = $this->findTestEnvVarInTravisYml($travisYmlFileContents);
+        if (empty($testVarEnvVar)) {
+            $output->writeln("<error>Cannot find test target environment variable in $travisYmlFile, skipping travis job addition.</error>");
+            return;
+        }
+
+        $testType = strtolower($testType);
+
+        $travisJobsToAdd = array();
+        if ($testType == 'ui') {
+            $travisJobsToAdd[] = "- TEST_SUITE=UITests MYSQL_ADAPTER=PDO_MYSQL TEST_AGAINST_PIWIK_BRANCH=\$$testVarEnvVar";
+        } else {
+            $travisJobsToAdd[] = "- TEST_SUITE=PluginTests MYSQL_ADAPTER=PDO_MYSQL TEST_AGAINST_PIWIK_BRANCH=\$$testVarEnvVar";
+            $travisJobsToAdd[] = "- TEST_SUITE=PluginTests MYSQL_ADAPTER=PDO_MYSQL TEST_AGAINST_CORE=minimum_required_piwik";
+        }
+
+        $jobAdded = false;
+        foreach ($travisJobsToAdd as $line) {
+            if (strpos($travisYmlFileContents, $line) !== false) {
+                continue;
+            }
+
+            if (!$this->doesMatrixSectionExist($travisYmlFileContents)) {
+                $travisYmlFileContents = $this->addInitialEnvMatrixWithJob($travisYmlFileContents, $line);
+            } else if ($this->isMatrixEnvSectionEmpty($travisYmlFileContents)) {
+                $travisYmlFileContents = $this->addJobToEmptyMatrix($travisYmlFileContents, $line);
+            } else {
+                $travisYmlFileContents = $this->addJobToNonEmptyMatrix($travisYmlFileContents, $line);
+            }
+
+            $jobAdded = true;
+        }
+
+        if ($jobAdded) {
+            file_put_contents($travisYmlFile, $travisYmlFileContents);
+        }
+    }
+
+    private function findTestEnvVarInTravisYml($travisYmlFileContents)
```

Comment:
Can we extract those private methods in a different class, make them public and tests for them?